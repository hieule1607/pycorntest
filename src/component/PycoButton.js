import React, { Component } from 'react';
import { View, TouchableOpacity, StyleSheet } from 'react-native';
import SlIcon from 'react-native-vector-icons/SimpleLineIcons';
import AntIcon from 'react-native-vector-icons/AntDesign';

export const CustomButton = (props) => {

  const { isSelected, iconName = 'lock', onPress } = props;

  return (
    <View>	
      <AntIcon name="caretup" size={10} color={isSelected ? 'green' : 'white'} style={{marginBottom: -5, alignSelf:'center'}}/>
      <View style={[isSelected ? styles.enableView : styles.disableView, {height: 2, marginHorizontal: 4}]}>										
      </View>
      <TouchableOpacity style={{padding: 10}} activeOpacity={1}
        onPress={onPress}>
        <SlIcon name={iconName} size={30} color={isSelected ? 'green' : 'rgb(217, 217, 217)'} />
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  enableView: {
    backgroundColor: 'green'
  },
  disableView: {
    backgroundColor: 'white'
  },
})