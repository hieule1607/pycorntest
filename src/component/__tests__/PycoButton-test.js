import React from 'react';
import renderer from 'react-test-renderer';
import PycoButton from "../PycoButton";

describe(`PycoButton Component`, () => {
    let props
    beforeEach(()=>{
        props = {
            iconName: 'lock', 
            isSelected: 4
        }
    })

    it(`should render correctly`, () => {
        const tree = renderer.create(<PycoButton {...props}/>).toJSON()
        expect(tree).toMatchSnapshot();
    })

})