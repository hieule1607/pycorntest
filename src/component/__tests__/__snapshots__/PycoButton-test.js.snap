// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PycoButton Component should render correctly 1`] = `
<View>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "green",
          "fontSize": 10,
        },
        Object {
          "alignSelf": "center",
          "marginBottom": -5,
        },
        Object {
          "fontFamily": "anticon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "green",
        },
        Object {
          "height": 2,
          "marginHorizontal": 4,
        },
      ]
    }
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "padding": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "green",
            "fontSize": 30,
          },
          undefined,
          Object {
            "fontFamily": "simple-line-icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
